### PR TITLE
Improve snapshot error messages

### PR DESCRIPTION
**Pull request by kwagyeman**
Make error messages for snapshot more human readable.

### CODE REVIEW

**Comment by iabdalkader** [comment]
Okay but might as well use full words (like Pixel format and Frame size etc..)... Also this will not stop questions like #1350 if it just says "Framebuffer Error" still doesn't explain what's wrong, best to add some more sanity checks in py_sensor_set_windowing().

**Comment by kwagyeman** [comment]
Error number -3 shouldn't happen ever. I don't know how to make the message more useful given that.

**Comment by iabdalkader** [comment]
Sorry was looking at another error number, the message is okay, if it's not possible to detect the error in py_set_windowing then I guess it's fine.

**Comment by kwagyeman** [comment]
I don't have all the information until snapshot is called. User could set the pixelformat after setwindowing.

File: src/omv/modules/py_sensor.c [hunk 1]
```diff
@@ -115,10 +115,21 @@ static mp_obj_t py_sensor_snapshot(uint n_args, const mp_obj_t *args, mp_map_t *
...
-    // Note: OV2640 JPEG mode can __fatal_error().
     int ret = sensor.snapshot(&sensor, (image_t *) py_image_cobj(image), 0);
 
-    if (ret < 0) {
+    if (ret == -1) {
+        mp_raise_msg(&mp_type_RuntimeError,
+                     MP_ERROR_TEXT("PIXFORMAT not set!"));
+    } else if (ret == -2) {
+        mp_raise_msg(&mp_type_RuntimeError,
+                     MP_ERROR_TEXT("FRAMESIZE not set, unsupported FRAMESIZE, or unsupported WINDOW!"));
```
Comment (iabdalkader) [inline src/omv/modules/py_sensor.c:125]:
```suggestion
                     MP_ERROR_TEXT("Invalid Frame Size or Window!"));
```

File: src/omv/modules/py_sensor.c [hunk 2]
```diff
@@ -115,10 +115,21 @@ static mp_obj_t py_sensor_snapshot(uint n_args, const mp_obj_t *args, mp_map_t *
 #endif // MICROPY_PY_IMU
 
     mp_obj_t image = py_image(0, 0, 0, 0);
-    // Note: OV2640 JPEG mode can __fatal_error().
     int ret = sensor.snapshot(&sensor, (image_t *) py_image_cobj(image), 0);
 
-    if (ret < 0) {
+    if (ret == -1) {
+        mp_raise_msg(&mp_type_RuntimeError,
+                     MP_ERROR_TEXT("PIXFORMAT not set!"));
```
Comment (iabdalkader) [inline src/omv/modules/py_sensor.c:122]: @kwagyeman I forgot to fix this, if you want to send another PR that's fine.
